driver_test.go

```diff
@@ -338,3 +309,302 @@ func TestIsDdl(t *testing.T) {
+				(3, "three", CAST("three" as bytes), 42, 42, true )`,
+			input: `UPDATE TestExecContextDml SET testSTring = "updated" WHERE key = 2 OR key = 3`,
+			query: `SELECT * FROM TestExecContextDml ORDER BY key`,
+			wantRows: []testExecContextDmlRow{
+				{key: 1, testString: "one", testBytes: []byte("one"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 2, testString: "updated", testBytes: []byte("two"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 3, testString: "updated", testBytes: []byte("three"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name: "update primary key",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true )`,
+			input:     `UPDATE TestExecContextDml SET key = 100 WHERE key = 1`,
+			wantError: true,
+		},
+		{
+			name: "update nothing",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true ),
+				(3, "three", CAST("three" as bytes), 42, 42, true )`,
+			input: `UPDATE TestExecContextDml SET testSTring = "nothing" WHERE false`,
+			query: `SELECT * FROM TestExecContextDml ORDER BY key`,
+			wantRows: []testExecContextDmlRow{
+				{key: 1, testString: "one", testBytes: []byte("one"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 2, testString: "two", testBytes: []byte("two"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 3, testString: "three", testBytes: []byte("three"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name: "update everything",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true ),
+				(3, "three", CAST("three" as bytes), 42, 42, true )`,
+			input: `UPDATE TestExecContextDml SET testSTring = "everything" WHERE true `,
+			query: `SELECT * FROM TestExecContextDml WHERE testString = "everything"`,
+			wantRows: []testExecContextDmlRow{
+				{key: 1, testString: "everything", testBytes: []byte("one"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 2, testString: "everything", testBytes: []byte("two"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 3, testString: "everything", testBytes: []byte("three"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name:      "update to wrong type",
+			input:     `UPDATE TestExecContextDml SET testBool = 100 WHERE key = 1`,
+			wantError: true,
+		},
+	}
+
+	// Run tests.
+	for _, tc := range tests {
+
+		// Set up test table.
+		if tc.setUp != "" {
+			if _, err = db.ExecContext(ctx, tc.setUp); err != nil {
+				t.Error(err)
```

Comment:
While in the test loop, these `t.Error` statements need to include the test name similar to the formatted errors below so it's clear which test is causing the error. Please fix.

Reply:
fixed for the 3 new errs

Reply:
Please also fix this one :)